src/overloads.ts

```diff
@@ -0,0 +1,182 @@
+import {StrictEqualUsingTSInternalIdenticalToOperator, IsNever, UnionToIntersection} from './utils'
+
+// prettier-ignore
+/**
+ * The simple(ish) way to get overload info from a function `F`. Recent versions of TypeScript will match any function against a generic 10-overload type, filling in slots with duplicates of the function.
+ * So, we can just match against a single type and get all the overloads.
+ *
+ * For older versions of TypeScript, we'll need to painstakingly do ten separate matches.
+ */
+export type TSPost53OverloadsInfoUnion<F> =
+  F extends {(...args: infer A1): infer R1; (...args: infer A2): infer R2; (...args: infer A3): infer R3; (...args: infer A4): infer R4; (...args: infer A5): infer R5; (...args: infer A6): infer R6; (...args: infer A7): infer R7; (...args: infer A8): infer R8; (...args: infer A9): infer R9; (...args: infer A10): infer R10}
+  ? ((...p: A1) => R1) | ((...p: A2) => R2) | ((...p: A3) => R3) | ((...p: A4) => R4) | ((...p: A5) => R5) | ((...p: A6) => R6) | ((...p: A7) => R7) | ((...p: A8) => R8) | ((...p: A9) => R9) | ((...p: A10) => R10)
+  : never
+
+export type UnknownFn = (...p: unknown[]) => unknown
+
+/**
+ * `true` iff `T` is equivalent to `(...p: unknown[]) => unknown`, which is what an overload variant looks like for a non-existent overload
```

Comment:
```suggestion
 * `true` if `T` is equivalent to `(...p: unknown[]) => unknown`, which is what an overload variant looks like for a non-existent overload.
```

Reply:
`iff` wasn't a typo - it means "if and only if". With just if it'd allow for it to be true for other types too